### PR TITLE
Avoid blank OAI xml elements

### PR DESCRIPTION
**Pull request by aaron-collier**
@jacobthill the issue with Texas Tech is that in at least one record they provide an empty publisher element. So we get something like:

```
<dc:publisher>Digitized by the Southwest Collection/Special Collections Library.</dc:publisher>
<dc:publisher/>
```

This PR checks if there are any keys once we flatten the value.

### CODE REVIEW

**Review by edsu — CHANGES_REQUESTED** [review]
I thought that [this check](https://github.com/sul-dlss/dlme-airflow/blob/834b75778e5482f32c59387e25f0c65fcd057bf3/dlme_airflow/drivers/oai_xml.py#L205) would have ensured that value was defined and not an empty string?

Would it be possible to have a test for this?

**Comment by aaron-collier** [comment]
@edsu yeah, I definitely need to address the tests for this. The issue I'm seeing with `get_tag` is that it's not providing the full tag. For instance, it returns `035` when we want `035_a` which happens in the flatten method, but originally that was after inserting into result, so we ended up with only the last result not all results when there were multiples. 

I'd be happy to chat about this if you have any better ideas :-)

File: dlme_airflow/drivers/oai_xml.py [hunk 1]
```diff
@@ -142,19 +142,24 @@ def _element_to_dict(self, rec_metadata):
...
-            if tag in result:
-                if isinstance(result[tag], str):
-                    if value is not None:
-                        result[tag] = [result[tag], value]
-                    else:
-                        result[tag] = [result[tag]]
-                elif value is not None:
-                    result[tag].append(value)
-            elif value is not None:
+            if value is None:
```
[the remark this answers is not in the document]
Comment (aaron-collier) [reply dlme_airflow/drivers/oai_xml.py:151]: Yeah, that's likely true, I'll work that out in a test.

**Comment by aaron-collier** [comment]
@edsu I think I've addressed your comments and added a bit of data to the example data to extend the test.

**Review by edsu — APPROVED** [review]

**Comment by edsu** [comment]
Thanks @aaron-collier!

**Pull request merged**